### PR TITLE
Feat: staking update

### PR DESCRIPTION
**Pull request by egor0798**
## Changes:
- Moved validator table to separate component
- Updated validators table styles
- Changed selected validators modal to show both selected and not selected validators

***

image
image

### CODE REVIEW

File: src/renderer/entities/staking/lib/useValidatorsMap.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+  const [era, setEra] = useState<number>();
+  const [validators, setValidators] = useState<ValidatorMap>({});
+
+  useEffect(() => {
+    let unsubEra: () => void | undefined;
+
+    if (api) {
+      (async () => {
+        unsubEra = await subscribeActiveEra(api, setEra);
```
Comment (tuul-wq) [inline src/renderer/entities/staking/lib/useValidatorsMap.ts:19]: `subscribeActiveEra(api, setEra).then(unsubFn => unsubEra = unsubFn);`
Would be a bit cleaner than `iife async await`

File: src/renderer/entities/staking/lib/useValidatorsMap.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    return () => {
+      api && unsubEra?.();
+    };
+  }, []);
+
+  useEffect(() => {
+    if (!era || !chainId || !api) return;
+
+    getValidatorsWithInfo(chainId, api, era, isLightClient).then((validators) => {
```
Comment (tuul-wq) [inline src/renderer/entities/staking/lib/useValidatorsMap.ts:31]: `getValidatorsWithInfo(chainId, api, era, isLightClient).then(setValidators)`

File: src/renderer/pages/Staking/Operations/components/Validators/Validators.test.tsx [hunk 3]
```diff
@@ -14,20 +23,14 @@ jest.mock('@renderer/app/providers', () => ({
...
-        totalStake: '84293898648764293',
-        ownStake: '27425315022592146',
-        blocked: false,
-      },
-    }),
   }),
   useEra: jest.fn().mockReturnValue({
     subscribeActiveEra: jest.fn().mockImplementation((api: any, eraCallback: any) => eraCallback(1)),
   }),
+  useValidatorsMap: jest.fn().mockImplementation(() => validatorsMap),
```
Comment (tuul-wq) [inline src/renderer/pages/Staking/Operations/components/Validators/Validators.test.tsx:33]: I guess this should work as well
`useValidatorsMap: jest.fn().mockResolvedValue(validatorsMap)`

File: src/renderer/pages/Staking/Operations/components/Validators/Validators.tsx [hunk 4]
```diff
@@ -36,38 +22,20 @@ type Props = {
...
-  const [era, setEra] = useState<number>();
-  const [validators, setValidators] = useState<ValidatorMap>({});
   const [isValidatorsLoading, setIsValidatorsLoading] = useState(true);
 
   const [query, setQuery] = useState('');
   const [maxValidators, setMaxValidators] = useState(0);
   const [selectedValidators, setSelectedValidators] = useState<Record<Address, boolean>>({});
 
   useEffect(() => {
-    let unsubEra: () => void | undefined;
```
Comment (tuul-wq) [inline src/renderer/pages/Staking/Operations/components/Validators/Validators.tsx:51]: 👍

File: src/renderer/pages/Staking/Operations/components/ValidatorsTable/ValidatorsTable.tsx [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+      <div className={cnTw('grid items-center pl-5 pr-2', TABLE_GRID_CELLS)}>
+        <FootnoteText className="text-text-secondary">{t('staking.validators.validatorTableHeader')}</FootnoteText>
+        <FootnoteText className="text-text-secondary px-3">{t('staking.validators.ownStakeTableHeader')}</FootnoteText>
+        <FootnoteText className="text-text-secondary px-3">
+          {t('staking.validators.totalStakeTableHeader')}
+        </FootnoteText>
+      </div>
+
+      <ul className="flex flex-col [overflow-y:overlay] max-h-[448px]">
+        {validators.map((validator) => children && children(validator, rowStyle))}
```
Comment (tuul-wq) [inline src/renderer/pages/Staking/Operations/components/ValidatorsTable/ValidatorsTable.tsx:32]: Why it checks `children` before calling it as function?
Props has `children` as required parameter

File: src/renderer/pages/Staking/Operations/components/ValidatorsTable/ValidatorsTable.tsx [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+import { cnTw, getComposedIdentity } from '@renderer/shared/lib/utils';
+import { BodyText, FootnoteText, HelpText, Icon, Identicon, InfoPopover } from '@renderer/shared/ui';
+import { AssetBalance } from '@renderer/entities/asset';
+import { AssetFiatBalance } from '@renderer/entities/price/ui/AssetFiatBalance';
+import { useI18n } from '@renderer/app/providers';
+import { ExplorerLink } from '@renderer/components/common';
+
+const TABLE_GRID_CELLS = 'grid-cols-[1fr,128px,128px,40px]';
+
+type Props = {
```
Comment (tuul-wq) [inline src/renderer/pages/Staking/Operations/components/ValidatorsTable/ValidatorsTable.tsx:11]: To be more specific as we do few lines below in `RowProps`
```suggestion
type TableProps = {
```

File: src/renderer/pages/Staking/Operations/components/ValidatorsTable/ValidatorsTable.tsx [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+};
+
+type RowProps = {
+  validator: Validator;
+  asset?: Asset;
+  explorers?: Explorer[];
+};
+
+const ValidatorRow = ({ validator, explorers, asset }: RowProps) => {
+  const getExplorers = (address: Address, explorers: Explorer[] = []) => {
```
Comment (tuul-wq) [inline src/renderer/pages/Staking/Operations/components/ValidatorsTable/ValidatorsTable.tsx:45]: Better to move this _pure_ function out of component
Add return type

File: src/renderer/pages/Staking/Operations/components/ValidatorsTable/ValidatorsTable.tsx [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+      </div>
+      <InfoPopover data={getExplorers(validator.address, explorers)} position="top-full right-0" buttonClassName="p-2">
+        <Icon name="info" size={16} className="mr-auto group-hover:text-icon-active" />
+      </InfoPopover>
+    </>
+  );
+};
+
+ValidatorsTable.Row = ValidatorRow;
+export default ValidatorsTable;
```
Comment (tuul-wq) [inline src/renderer/pages/Staking/Operations/components/ValidatorsTable/ValidatorsTable.tsx:91]: Why not named export?

**Review by tuul-wq — CHANGES_REQUESTED** [review]

**Review by Asmadek — APPROVED** [review]

File: src/renderer/pages/Staking/Operations/components/Validators/Validators.test.tsx (hunk 3 above)
Comment (egor0798) [reply src/renderer/pages/Staking/Operations/components/Validators/Validators.test.tsx:33]: No, it doesn't. It throws error abount using validatorsMap const before initialisation

File: src/renderer/pages/Staking/Operations/components/ValidatorsTable/ValidatorsTable.tsx (hunk 7 above)
Comment (egor0798) [reply src/renderer/pages/Staking/Operations/components/ValidatorsTable/ValidatorsTable.tsx:45]: moved to local utils

**Review by tuul-wq — APPROVED** [review]

**Pull request merged**